src/main/java/org/apache/skywalking/banyandb/v1/client/BanyanDBClient.java

```diff
@@ -373,7 +386,14 @@ private void defineIndexRules(Stream stream, List<IndexRule> indexRules) throws
 
         IndexRuleMetadataRegistry irRegistry = new IndexRuleMetadataRegistry(checkNotNull(this.channel));
         for (final IndexRule ir : indexRules) {
-            irRegistry.create(ir);
+            try {
+                irRegistry.create(ir);
+            } catch (BanyanDBException ex) {
+                if (ex.getStatus().equals(Status.Code.ALREADY_EXISTS)) {
+                    continue;
+                }
```

Comment:
In code practice, we should not use `exception` to control workflow. Please use `isExist` before `create`.

Reply:
> In code practice, we should not use `exception` to control workflow. Please use `isExist` before `create`.

Currently, we don't have `isExist` API.

Reply:
@hanahmily Could you track this? We should add one, this is not a good practice.

Reply:
> @hanahmily Could you track this? We should add one, this is not a good practice.

If we add `isExist`, is it possible we would introduce a potential data race?

Reply:
The current logic should be kept as it could reduce 99% exception cases, the lock mechanism has to exist due to avoid race and inconsistent data.

Reply:
We have a series of `getXXX` to check this. 

BTW, we can't avoid this exception under a concurrent creating scenario even get the resource before creating it. I tend to leave this piece as it is and add a getting operation before the creating.

Reply:
If we have `getxxx`, we should take it.

Reply:
> We have a series of `getXXX` to check this.
> 
> BTW, we can't avoid this exception under a concurrent creating scenario even get the resource before creating it. I tend to leave this piece as it is and add a getting operation before the creating.

IIRC `getXXX` will throw `NOT_FOUND` now if no schema is found...We have to check that exception too...